kevm_pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -351,17 +362,8 @@ def recur_mapping(syntax, lhs, rhs, var_idx, key_type_name, value_type_name):
 
 
 def generate_function_sentences(contract: Contract) -> List[KSentence]:
-
     function_call_data_production: KSentence = KProduction(KSort('ByteArray'), [KNonTerminal(contract.sort), KTerminal('.'), KNonTerminal(contract.sort_method)], klabel=contract.klabel_method, att=KAtt({'function': ''}))
-    function_sentence_pairs = _extract_function_sentences(contract)
-
-    function_productions: List[KSentence] = []
-    function_rules: List[KSentence] = []
-    for f_prod, f_rule in function_sentence_pairs:
-        function_productions.append(f_prod)
-        function_rules.append(f_rule)
-
-    function_sentences = function_productions + function_rules
+    function_sentences = contract.method_sentences
     return [function_call_data_production] + function_sentences if function_sentences else []
```

Comment:
`if function_sentences else []` looks redundant.

Reply:
This is tricky, but operator precedence is:

```python3
([function_call_data_production] + function_sentences) if function_sentences else []
```

Reply:
Yeah this is to keep from generating subsort productions if the supersort doesn't exist. Otherwise we end up with something like:

```
syntax Int ::= TokenContract "." TokenField
```

But where `TokenField` has not been declared, for example.

Reply:
OK, lets use parens around the first part to make this obvious. I can see someone else making the same mistake.

Reply:
This code got removed anyway.